modules/apps/headless/headless-admin-workflow/headless-admin-workflow-impl/src/main/java/com/liferay/headless/admin/workflow/internal/resource/v1_0/WorkflowDefinitionLinkResourceImpl.java

```diff
@@ -133,24 +138,77 @@ public WorkflowDefinitionLink postWorkflowDefinitionWorkflowDefinitionLink(
 				workflowDefinition.getName(), workflowDefinition.getVersion()));
 	}
 
+	@Override
+	public WorkflowDefinitionLink
+			putWorkflowDefinitionLinkByExternalReferenceCode(
+				String externalReferenceCode,
+				WorkflowDefinitionLink workflowDefinitionLink)
+		throws Exception {
+
+		long groupId = workflowDefinitionLink.getGroupId();
+
+		if (groupId != 0) {
```

Comment:
Is this logic correct? You check if groupId has a non-zero value, but you didn't use the value inside the if block.